crates/sui-transaction-builder/src/lib.rs

```diff
@@ -0,0 +1,1091 @@
+        let status = match effects.unwrap().unwrap() {
+            sui_types::types::TransactionEffects::V2(e) => Some(e.status),
+            _ => None,
+        };
+        let expected_status = ExecutionStatus::Success;
+        // The tx failed, so we expect Failure instead of Success
+        assert_ne!(expected_status, status.unwrap());
+    }
+
+    #[tokio::test]
+    async fn test_merge_coins() {
+        let client = Client::new_localhost();
+        let mut tx = TransactionBuilder::new();
+        let (address, pk, coins) = helper_setup(&mut tx, &client).await;
+
+        let coin1 = coins.first().unwrap().id;
+        let coin1_obj = client.object(coin1.into(), None).await.unwrap().unwrap();
+        let coin1_digest = coin1_obj.digest();
+        let coin1_version = coin1_obj.version();
+
+        let coin_to_merge = tx.input(unresolved::Input::owned(
+            coin1_obj.object_id(),
+            coin1_version,
+            coin1_digest,
+        ));
+
+        let mut coins_to_merge = vec![];
+        // last coin is used for gas, first coin is the one we merge into
+        for c in coins[1..&coins.len() - 1].iter() {
+            let coin = client.object(c.id.into(), None).await.unwrap().unwrap();
+            let coin_digest = coin.digest();
+            let coin_version = coin.version();
+            coins_to_merge.push(tx.input(unresolved::Input::owned(
+                coin.object_id(),
+                coin_version,
+                coin_digest,
+            )));
+        }
+
+        tx.merge_coins(coin_to_merge, coins_to_merge);
+        let tx = tx.finish().unwrap();
+        let sig = pk.sign_transaction(&tx).unwrap();
+
+        let effects = client.execute_tx(vec![sig], &tx).await;
+        wait_for_tx_and_check_effects_status_success(&client, &tx, effects).await;
+
+        // check that there are two coins
+        let coins_after = client
+            .coins(address, None, PaginationFilter::default())
+            .await
+            .unwrap();
+        assert_eq!(coins_after.data().len(), 2);
+    }
+
+    #[tokio::test]
+    async fn test_make_move_vec() {
+        let client = Client::new_localhost();
+        let mut tx = TransactionBuilder::new();
+        let (_, pk, _) = helper_setup(&mut tx, &client).await;
+
+        let input = tx.input(Serialized(&1u64));
+        tx.make_move_vec(Some(TypeTag::U64), vec![input]);
+
+        let tx = tx.finish().unwrap();
+        let sig = pk.sign_transaction(&tx).unwrap();
+
+        let effects = client.execute_tx(vec![sig], &tx).await;
+        wait_for_tx_and_check_effects_status_success(&client, &tx, effects).await;
+    }
+
+    #[tokio::test]
+    async fn test_publish() {
+        let client = Client::new_localhost();
+        let mut tx = TransactionBuilder::new();
+        let (address, pk, _) = helper_setup(&mut tx, &client).await;
+
+        // publish very dummy package (from sui/examples/move/first_package)
+        let modules: [u8; 506] = [
```

Comment:
I'm not against having raw compiled bytes here, if we do want to compile the bytes as a part of the test then we'd need to build utilities to shell out to the move compiler or take a dep on the mono-repo for that (which would be nice to avoid)

Reply:
To avoid duplicating code, we could use the `sui` binary to do a `sui move build --dump-bytecode-base64` and get the compiled modules, digest, and dependencies from there. 
The advantage is that we're not copy pasting code and we're already using the CLI to start a local network for tests, so this seems reasonable to me.

Reply:
Let me know 👍  or 👎

Reply:
I'm happy with the shelling out approach, IIUC you are already using a Makefile as a test runner, so you could use that to generate test fixtures into a directory before the test starts -- seems fine to me.

Reply:
I think we are saying the same thing, but just to be explicit, could those instructions be a make rule that builds the Move bytecode (at the make level we need to depend on the `sui` CLI anyway to run the network), and could the test's make rule depend on that rule to make sure the fixture is always up-to-date?

That way there is no shelling out in the Rust code for the test, no dependency on the mono-repo at the crate level, but people don't have to worry about understanding how the bytes are generated and manually updating them.

Reply:
Let's go like this and if needed down the road, happy to add another way to handle this.

Reply:
It's maybe less relevant for testing publishing and upgrading but how will you test making a move call in a PTB without being able to publish a package with functions to call?

Reply:
Unless we need to test specific hand-rolled packages, we can use the framework to call different functions?